src/ripple/ledger/ReadView.h

```diff
@@ -223,7 +223,7 @@ class ReadView
     {
     }
 
-    ReadView (ReadView&& other)
+    ReadView (ReadView&& other) noexcept
```

Comment:
Need to make these statements compile:
```c++
        static_assert(std::is_nothrow_constructible<decltype(sles), ReadView&>{}, "");
        static_assert(std::is_nothrow_constructible<decltype(txs), ReadView&>{}, "");
```

Reply:
Will work on that. Thanks! I'm also in progress of adding the usual `static_assert` checks like you mentioned earlier. For most user-defined-types, in other codebases, I usually just write `static_asserts` for all ctor, dtor, copy, copy assign, move, and move-assign (with noexcept) just to be overly explicit. Happy to adopt that same philosophy in this codebase.

Reply:
To be honest, we haven't been adopting this practice universally.  But I wouldn't object to it in test code.  And the occasional place like this where you need to make sure a `= default` is doing the right thing isn't a bad idea either.

Reply:
Fair enough. I'm in the (slow) progress of being a lot more careful with this PR.  I'm making small commits and marking them with the usual `[FOLD]` so you can follow the foot-trail I leave behind. I'll ping you when I think it's in a good reviewable state again. Until then, I'll probably just close the PR and re-open when it's ready.

Reply:
Thanks much for you work on this.  I think this will be a valuable addition.